### PR TITLE
Add 'prefixText' and 'suffixText' when renaming shorthand properties

### LINKED ISSUE #12007: Renaming interface members and object shorthand property names
**TypeScript Version:**   nightly (2.1.0-dev.20161102)

**Code**

```ts
interface Test {
    foo: string
}


function bar(): Test {
    const foo = 'foo'
    return { foo }
}
```

Rename `foo` on line `2` to `foo2`

**Expected behavior:**
```ts
interface Test {
    foo2: string
}


function bar(): Test {
    const foo = 'foo'
    return { foo2: foo }
}
```

**Actual behavior:** 

```ts
interface Test {
    foo2: string
}


function bar(): Test {
    const foo = 'foo'
    return { foo2 } // <------- broken code
}
```

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #12007

### CODE REVIEW

File: src/harness/harnessLanguageService.ts [hunk 1]
```diff
@@ -459,7 +459,9 @@ namespace Harness.LanguageService {
             return unwrapJSONCallResult(this.shim.getRenameInfo(fileName, position));
         }
         findRenameLocations(fileName: string, position: number, findInStrings: boolean, findInComments: boolean): ts.RenameLocation[] {
-            return unwrapJSONCallResult(this.shim.findRenameLocations(fileName, position, findInStrings, findInComments));
+            const res = unwrapJSONCallResult(this.shim.findRenameLocations(fileName, position, findInStrings, findInComments)) as ReadonlyArray<ts.RenameLocation>;
+            // JSON serialization removes undefined properties, so add them back if necessary.
+            return res.map((r): ts.RenameLocation => ({ prefixText: undefined, suffixText: undefined, ...r }));
```
Comment (sheetalkamat) [inline src/harness/harnessLanguageService.ts]: why? Not having those properties should result in same result as having them as undefined?

File: src/server/protocol.ts [hunk 2]
```diff
@@ -1135,7 +1135,12 @@ namespace ts.server.protocol {
         /** The file to which the spans apply */
         file: string;
         /** The text spans in this group */
-        locs: TextSpan[];
+        locs: RenameTextSpan[];
+    }
+
+    export interface RenameTextSpan extends TextSpan {
+        readonly prefixText: string | undefined;
```
Comment (sheetalkamat) [inline src/server/protocol.ts]: Can you make these optional instead?

**Comment by andy-ms** [comment]
@sheetalkamat Good to merge?

**Comment by dhlolo** [comment]
What if I want to call tsLanguageService.findRenameLocations(...) to rename the key of short hand property? It seems that I cannot use this api to achieve my goal right now.

**Pull request merged**